packages/mobile/android/app/build.gradle

```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion flutter.compileSdkVersion
+    compileSdkVersion 31
```

Comment:
Why was this change required?

Reply:
It was specified in the documentation of the geolocator plugin to set the compileSdkVersion as 31.